internal/external/procfile/generate.go

```diff
@@ -0,0 +1,48 @@
+package procfile
+
+import (
+	"bytes"
+
+	"gopkg.in/alessio/shellescape.v1"
+	"mvdan.cc/sh/v3/syntax"
+)
+
+func Generate(procs []Process) (string, error) {
```

Comment:
Preferable to operate on an io.Writer here, as that's the most general thing:

```suggestion
func Generate(w io.Writer, procs []Process) error {
```